liblocation/src/androidTest/java/com/mapbox/android/core/location/GoogleLocationEngineInstrumentedTest.java

```diff
@@ -0,0 +1,115 @@
+package com.mapbox.android.core.location;
+
+import android.location.Location;
+import android.os.Looper;
+import android.support.annotation.NonNull;
+import android.support.test.runner.AndroidJUnit4;
+import com.google.android.gms.location.FusedLocationProviderClient;
+import com.google.android.gms.location.LocationCallback;
+import com.google.android.gms.location.LocationRequest;
+import com.google.android.gms.location.LocationResult;
+import com.google.android.gms.tasks.OnSuccessListener;
+
+import com.google.android.gms.tasks.Task;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(AndroidJUnit4.class)
+public class GoogleLocationEngineInstrumentedTest {
```

Comment:
same here we should not be mocking fused provider and perhaps start supporting https://developers.google.com/android/reference/com/google/android/gms/location/FusedLocationProviderClient.html#setMockLocation(android.location.Location) for testing purposes.